integration/system/options_test.go

```diff
@@ -0,0 +1,24 @@
+package system
+
+import (
+	"net/http"
+	"testing"
+
+	req "github.com/docker/docker/internal/test/request"
+	"gotest.tools/assert"
+)
+
+func TestAPIOptionsRoute(t *testing.T) {
+	defer setupTest(t)()
+
+	resp, _, err := req.Do("/", req.Method(http.MethodOptions))
+	assert.NilError(t, err)
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+}
+
+func TestAPIGetEnabledCORS(t *testing.T) {
```

Comment:
Looks like this test is currently doing anything, and it was originally written to test that the CORS headers were added (which defaulted to `*`); also see my comment above.

This is the commit that added the original version of the test (it moved around a few times); https://github.com/moby/moby/commit/6d5bdff3942ce5e030b2cbd1510f418de25a1a53, which was part of https://github.com/moby/moby/pull/806


Pending the options in my earlier comment, the `if` should be removed, or otherwise this test should start a new daemon with the correct flag set.

Removing the `if`, or starting the daemon with `--api-cors-header='*'` both give;

```bash
curl --unix-socket /var/run/docker.sock http://local/_ping -v
*   Trying /var/run/docker.sock...
* Connected to local (/var/run/docker.sock) port 80 (#0)
> GET /_ping HTTP/1.1
> Host: local
> User-Agent: curl/7.52.1
> Accept: */*
> 
< HTTP/1.1 200 OK
< Access-Control-Allow-Headers: Origin, X-Requested-With, Content-Type, Accept, X-Registry-Auth
< Access-Control-Allow-Methods: HEAD, GET, POST, DELETE, PUT, OPTIONS
< Access-Control-Allow-Origin: *
< Api-Version: 1.41
< Cache-Control: no-cache, no-store, must-revalidate
< Docker-Experimental: false
< Ostype: linux
< Pragma: no-cache
< Server: Docker/dev (linux)
< Date: Mon, 19 Aug 2019 09:51:33 GMT
< Content-Length: 2
< Content-Type: text/plain; charset=utf-8
< 
* Curl_http_done: called premature == 0
* Connection #0 to host local left intact
```